pkg/sql/compile/compile.go

```diff
@@ -472,6 +471,21 @@ func (c *Compile) compilePlanScope(n *plan.Node, ns []*plan.Node) ([]*Scope, err
 			return nil, err
 		}
 		return ss, nil
+	case plan.Node_UNNEST:
```

Comment:
Is there anything special about unset. Why introduce a new type?

Reply:
Unnest is a table function operator, I find there is no table function supportted before, so I add this type.

Reply:
The operator can generate the table, but I don't understand why I need to add a new type. My understanding is that as a normal operator, adding an opeator is sufficient.

Reply:
In the simplest way, you can make two scans, like this example:
```sql
select * from t1, unnest(t1.a, path) as u.
```
It can be made into a plan like this:
```
scan(t1) join unnnsert(scan(t1.a), path)

Reply:
Thank you for your nice advice. I have another question, if no unnest type in plan, how can I tag a unnest in buildPlan function, and do the next step things, such as append vm.unnest operator in scope.

Reply:
Or which already exists type should I use.

Reply:
The github communication feels a bit inefficient. How about we email or wechat or something like that, it's faster.

Reply:
> Thank you for your nice advice. I have another question, if no unnest type in plan, how can I tag a unnest in buildPlan function, and do the next step things, such as append vm.unnest operator in scope.

Yes, you can, just add a node type. My understanding unsert can introduce a node type. Go for it, nice work.

Reply:
> Or which already exists type should I use.

Multiplexing project?My understanding is that it is enough to do an operation after projection.

Reply:
or use function:
```
(select * from t1) join (select unnest(t1.a, path) from t1)